faucet/valve.py

```diff
@@ -899,11 +901,21 @@ def _lacp_actions(self, lacp_pkt, port, now):
                     down_state = lacp_peer
                 elif lacp_peer.dyn_lldp_beacon_recv_state == STACK_STATE_FAILOVER:
                     failover_state = lacp_peer
+                lacp_peer_dp = None
+                if lacp_peer.lldp_peer_mac in self.dp.dps_by_mac:
+                    lacp_peer_dp = self.dp.dps_by_mac[lacp_peer.lldp_peer_mac]
+                if lacp_peer_dp:
+                    no_switch_up = no_switch_up and not lacp_peer_dp.dyn_running
+                    if not (lacp_down or lldp_down) and not lacp_peer_dp.dyn_running:
+                        crosslink_up = True
             if down_state:
                 if not failover_state:
-                    self.logger.warning('Suppressing LACP LAG %s on %s, peer %s is down' %
-                                        (port.lacp, port, down_state))
-                    return []
+                    if not (crosslink_up and no_switch_up):
+                        self.logger.warning('Suppressing LACP LAG %s on %s, peer %s is down' %
+                                (port.lacp, port, down_state))
+                        return []
+                    else:
+                        self.logger.warning('Not suppressing LACP LAG %s on %s. T2 switch is down. ' %(port.lacp, port))
```

Comment:
there is no concept of "T2" here -- just local-peer switch is down

and missing space after %